armi/reactor/tests/test_hexBlockRotate.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2024 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the ability to rotate a hexagonal block."""
+
+import copy
+import math
+import unittest
+
+import numpy as np
+
+from armi.reactor.blocks import HexBlock
+from armi.reactor.components import Component
+from armi.reactor.grids import (
+    MultiIndexLocation,
+    CoordinateLocation,
+    IndexLocation,
+    HexGrid,
+)
+from armi.reactor.tests.test_blocks import loadTestBlock, NUM_PINS_IN_TEST_BLOCK
+from armi.utils import iterables
+
+
+class HexBlockRotateTests(unittest.TestCase):
+    """Tests for various rotation aspects of a hex block."""
+
+    BOUNDARY_PARAMS = [
+        "cornerFastFlux",
+        "pointsCornerDpa",
+        "pointsCornerDpaRate",
+        "pointsCornerFastFluxFr",
+        "pointsEdgeDpa",
+        "pointsEdgeDpaRate",
+        "pointsEdgeFastFluxFr",
+        "THedgeTemp",
+        "THcornTemp",
+    ]
+    BOUNDARY_DATA = np.arange(6, dtype=float) * 10
+
+    PIN_PARAMS = [
+        "percentBuByPin",
+        "linPowByPin",
+    ]
+
+    PIN_DATA = np.arange(NUM_PINS_IN_TEST_BLOCK, dtype=float)
+
+    @classmethod
+    def setUpClass(cls):
+        cls.BASE_BLOCK = loadTestBlock()
```

Comment:
So... if this is in `setUpClass(cls)` and not `setUp(self)`, then you think ALL tests should share the same, class-level block?

But... what if I'm running the tests in parallel, and each test rotates the block? Then... depending on the order the tests are run in, the block might get rotated 4 times during one test?

I understand your tests are passing on GitHub, I can see that. 

But I think this should be a `setUp(self)` declaration, because if each test rotates a block once, I think we want to conceptually keep that block in question unique to each test.

IMO